src/DataTypes/array.tsx

```diff
@@ -0,0 +1,161 @@
+import type { JSONSchema7 as JSONSchema } from 'json-schema';
+import {
+	FULL_TEXT_SLUG,
+	createModelFilter,
+	type FormData,
+} from '../components/Filters/SchemaSieve';
+import { CreateFilter, getDataTypeSchema } from './utils';
+import { getDataModel } from '.';
+import { isJSONSchema, getRefSchema } from '../AutoUI/models/helpers';
+
+export const operators = () => ({
+	contains: 'contains',
+	not_contains: 'does not contain',
+});
+
+export type OperatorSlug =
+	| keyof ReturnType<typeof operators>
+	| typeof FULL_TEXT_SLUG;
+
+const isSchemaWithPrimitiveItems = (
+	schema: JSONSchema,
+): schema is JSONSchema & { items: JSONSchema } =>
+	!!schema?.items &&
+	typeof schema.items !== 'boolean' &&
+	'type' in schema.items;
+
+const isArrayOfObjectSchema = (schema: JSONSchema): boolean => {
+	return (
+		!!schema?.items &&
+		typeof schema.items !== 'boolean' &&
+		'properties' in schema.items &&
+		!!schema.items.properties
+	);
+};
+
+const buildFilterForPropertySchema = (
+	field: string,
+	operator: OperatorSlug,
+	value: string,
+	schema: JSONSchema,
+): JSONSchema | {} => {
+	const filter = getFilter(field, schema, value, operator);
+	if (!Object.keys(filter).length) {
```

Comment:
Why do we return `{}` in `getFilters` instead of just a nullish value?

Reply:
as we need to check in the create filter if the filter respect a JSONSchema logic, it is simpler and safer to just return empty object as we do for all DataTypes. This avoid a check and cover us from code exploding when checking with ajv

https://github.com/balena-io-modules/autoui/blob/274c775fc42bb046309fb4bd2bb9a98878a5fcdd/src/components/Filters/SchemaSieve.ts#L136